### PR TITLE
feat(azure): New checks related to PostgreSQL service 

### PR DESCRIPTION
**Pull request by pedrooot**
### Context
This pr adds 7 new checks related to PostgreSQL service

### Description

The new checks are:

- [x] postgresql_flexible_server_enforce_ssl_enabled
- [x] postgresql_flexible_server_log_checkpoints_on
- [x] postgresql_flexible_server_log_connections_on
- [x] postgresql_flexible_server_log_disconnections_on
- [x] postgresql_flexible_server_connection_throttling_on
- [x] postgresql_flexible_server_log_retention_days_greater_3
- [x] postgresql_flexible_server_allow_access_services_disabled

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "prowler",
+  "lockfileVersion": 3,
+  "requires": true,
+  "packages": {}
+}
```
Comment (MrCloudSec) [inline package-lock.json:6]: Please, remove this file.

File: prowler/providers/azure/services/postgresql/postgresql_flexible_server_enforce_ssl_enabled/postgresql_flexible_server_enforce_ssl_enabled.metadata.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  "ResourceType": "PostgreSQL",
+  "Description": "Enable SSL connection on PostgreSQL Servers.",
+  "Risk": "SSL connectivity helps to provide a new layer of security by connecting database server to client applications using Secure Sockets Layer (SSL). Enforcing SSL connections between database server and client applications helps protect against 'man in the middle' attacks by encrypting the data stream between the server and application.",
+  "RelatedUrl": "https://learn.microsoft.com/en-us/azure/postgresql/single-server/concepts-ssl-connection-security",
+  "Remediation": {
+    "Code": {
+      "CLI": "az postgres server update --resource-group <resourceGroupName> --name <serverName> --ssl-enforcement Enabled",
+      "NativeIaC": "",
+      "Other": "https://docs.bridgecrew.io/docs/bc_azr_networking_10",
+      "Terraform": ""
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/postgresql/postgresql_flexible_server_enforce_ssl_enabled/postgresql_flexible_server_enforce_ssl_enabled.metadata.json:19]:
```suggestion
      "Terraform": "https://docs.bridgecrew.io/docs/bc_azr_networking_10#terraform"
```

File: prowler/providers/azure/services/postgresql/postgresql_flexible_server_enforce_ssl_enabled/postgresql_flexible_server_enforce_ssl_enabled.metadata.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  "Severity": "medium",
+  "ResourceType": "PostgreSQL",
+  "Description": "Enable SSL connection on PostgreSQL Servers.",
+  "Risk": "SSL connectivity helps to provide a new layer of security by connecting database server to client applications using Secure Sockets Layer (SSL). Enforcing SSL connections between database server and client applications helps protect against 'man in the middle' attacks by encrypting the data stream between the server and application.",
+  "RelatedUrl": "https://learn.microsoft.com/en-us/azure/postgresql/single-server/concepts-ssl-connection-security",
+  "Remediation": {
+    "Code": {
+      "CLI": "az postgres server update --resource-group <resourceGroupName> --name <serverName> --ssl-enforcement Enabled",
+      "NativeIaC": "",
+      "Other": "https://docs.bridgecrew.io/docs/bc_azr_networking_10",
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/postgresql/postgresql_flexible_server_enforce_ssl_enabled/postgresql_flexible_server_enforce_ssl_enabled.metadata.json:18]: Is there a TrendMicro link for this?

File: prowler/providers/azure/services/postgresql/postgresql_service.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+                            resource_group=resource_group,
+                            require_secure_transport=require_secure_transport,
+                            log_checkpoints=log_checkpoints,
+                            log_connections=log_connections,
+                            log_disconnections=log_disconnections,
+                            connection_throttling=connection_throttling,
+                            log_retention_days=log_retention_days,
+                            firewall=self.__get_firewall__(
+                                subscription, resource_group, postgresql_server.name
+                            ),
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/postgresql/postgresql_service.py:54]: Can you set it up like the other ones, please?

File: pyproject.toml [hunk 5]
```diff
@@ -29,6 +29,7 @@ alive-progress = "3.1.5"
 awsipranges = "0.3.3"
 azure-identity = "1.15.0"
 azure-mgmt-authorization = "4.0.0"
+azure-mgmt-rdbms = "^10.1.0"
```
Comment (MrCloudSec) [inline pyproject.toml:32]:
```suggestion
azure-mgmt-rdbms = "10.1.0"
```

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Please, check my comments.

File: package-lock.json (hunk 1 above)
Comment (pedrooot) [reply package-lock.json:6]: Done!

File: prowler/providers/azure/services/postgresql/postgresql_flexible_server_enforce_ssl_enabled/postgresql_flexible_server_enforce_ssl_enabled.metadata.json (hunk 2 above)
Comment (pedrooot) [reply prowler/providers/azure/services/postgresql/postgresql_flexible_server_enforce_ssl_enabled/postgresql_flexible_server_enforce_ssl_enabled.metadata.json:19]: Done!

File: prowler/providers/azure/services/postgresql/postgresql_flexible_server_enforce_ssl_enabled/postgresql_flexible_server_enforce_ssl_enabled.metadata.json (hunk 3 above)
Comment (pedrooot) [reply prowler/providers/azure/services/postgresql/postgresql_flexible_server_enforce_ssl_enabled/postgresql_flexible_server_enforce_ssl_enabled.metadata.json:18]: No, TrendMicro is not available for this one

File: prowler/providers/azure/services/postgresql/postgresql_service.py (hunk 4 above)
Comment (pedrooot) [reply prowler/providers/azure/services/postgresql/postgresql_service.py:54]: Done!

File: pyproject.toml (hunk 5 above)
Comment (pedrooot) [reply pyproject.toml:32]: Done!

**Comment by pedrooot** [comment]
Updated to: https://github.com/prowler-cloud/prowler/pull/3409

**Pull request closed without merging**